src/Utilities/Compiler/PooledObjects/TemporarySet`1.cs

```diff
@@ -74,11 +74,27 @@ public bool Add(T item, CancellationToken cancellationToken)
             return storage.Contains(item);
         }
 
+        public readonly bool Any_NonConcurrent()
+        {
+            if (_storage is not { } storage)
+                return false;
+
+            return storage.Count > 0;
+        }
+
         public readonly Enumerator GetEnumerator_NonConcurrent()
         {
             return new Enumerator((_storage ?? EmptyHashSet).GetEnumerator());
         }
 
+        public readonly IEnumerable<T> AsEunumerable_NonConcurrent()
+        {
+            foreach (var item in _storage ?? EmptyHashSet)
```

Comment:
```suggestion
            foreach (var item in NonConcurrentEnumerable)
```

Reply:
Actually I think I'd rather method just return directly:

```
return (_storage ?? EmptyHashSet).GetEnumerator();
```

Reply:
Needs to be `return (_storage ?? EmptyHashSet).AsEnumerable();`. Done